src/Compilers/CSharp/Portable/FlowAnalysis/CSharpDataFlowAnalysis.cs

```diff
@@ -101,6 +102,46 @@ public override ImmutableArray<ISymbol> DataFlowsIn
             }
         }
 
+        /// <summary>
+        /// The set of local variables which are definitely assigned a value when a region is
+        /// entered.
+        /// </summary>
+        public override ImmutableArray<ISymbol> DefinitelyAssignedOnEntry
+        {
+            get
+            {
+                if (_definitelyAssignedOnEntry.IsDefault)
+                {
+                    var entryResult = ImmutableArray<ISymbol>.Empty;
+                    var exitResult = ImmutableArray<ISymbol>.Empty;
+                    if (Succeeded)
+                    {
+                        var (entry, exit) = DefinitelyAssignedWalker.Analyze(_context.Compilation, _context.Member, _context.BoundNode, _context.FirstInRegion, _context.LastInRegion);
+                        entryResult = Normalize(entry);
+                        exitResult = Normalize(exit);
+                    }
+
+                    ImmutableInterlocked.InterlockedInitialize(ref _definitelyAssignedOnEntry, entryResult);
+                    ImmutableInterlocked.InterlockedInitialize(ref _definitelyAssignedOnExit, exitResult);
+                }
+
+                return _definitelyAssignedOnEntry;
+            }
+        }
+
+        /// <summary>
+        /// The set of local variables which are definitely assigned a value when a region is
+        /// exited.
+        /// </summary>
+        public override ImmutableArray<ISymbol> DefinitelyAssignedOnExit
+        {
+            get
+            {
+                _ = DefinitelyAssignedOnEntry;
```

Comment:
We should not be executing properties for side effects. I would split this logic into a helper that calculates both, and then call that helper in both properties.

Reply:
>We should not be executing properties for side effects. 

Not that i disagree, but this is how this class works for all the other properties.  For example:

```c#
        public sealed override bool Succeeded
        {
            get
            {
                if (_succeeded == null)
                {
                    var discarded = DataFlowsIn;
                }

                return _succeeded.Value;
            }
        }

        public override ImmutableArray<ISymbol> DataFlowsOut
        {
            get
            {
                var discarded = DataFlowsIn; // force DataFlowsIn to be computed
                if (_dataFlowsOut.IsDefault)
                {
```

etc. etc.

I was following a when-in-rome approach here.  LMK if you still do want me to change this, or if you're ok with keeping the code as is.  ThankS!

Reply:
We should still do it here. That code should be changed as well

Reply:
Ok, have done for my code :D